settingsmetadata/controllers/SettingsMetadataController.go

```diff
@@ -0,0 +1,276 @@
+	if err != nil {
+		context.LogError(err, microappLog.MessageUnableToFindURLResource)
+		microappWeb.RespondError(w, err)
+		return
+	}
+
+	if err := controller.checkAndInitializeSettingsMetadata(); err != nil {
+		context.LogError(err, fmt.Sprintf(microappLog.MessageGenericErrorTemplate, "initializing settings-metadata"))
+		microappWeb.RespondError(w, err)
+		return
+	}
+
+	settingsmetadata := controller.settingsMetadatas
+	if tenantID.String() == "00000000-0000-0000-0000-000000000000" {
+		settingsmetadata = controller.globalsettingsMetadatas
+	}
+
+	tenant, err := controller.getTenant(context, uow, controller.repository, tenantID)
+	if err != nil {
+		context.LogError(err, fmt.Sprintf(microappLog.MessageGenericErrorTemplate, "getting tenant from database"))
+		microappWeb.RespondError(w, err)
+		return
+	}
+
+	err = tenant.GetTenantSettings(settingsmetadata)
+	if err != nil {
+		context.LogError(err, fmt.Sprintf(microappLog.MessageGenericErrorTemplate, "getting tenant settings from database"))
+		microappWeb.RespondError(w, err)
+		return
+	}
+
+	microappWeb.RespondJSON(w, http.StatusOK, toDTO(tenant))
+}
+
+func (controller *SettingsMetadataController) update(w http.ResponseWriter, r *http.Request, token *microappSecurity.JwtToken) {
+	context := controller.app.NewExecutionContext(token, microapp.GetCorrelationIDFromRequest(r), "tenantsettings.update", true, true)
+	uow := context.GetUOW()
+	defer uow.Complete()
+	params := mux.Vars(r)
+	stringTenantID := params["id"]
+	var reqDTO tenantDTO
+	if err := microappWeb.UnmarshalJSON(r, &reqDTO); err != nil {
+		context.LogJSONParseError(err)
+		microappWeb.RespondError(w, err)
+		return
+	}
+
+	tenantID, err := tenantService.GetTenantIDFromToken().GetTenantIDAsUUID(mux.Vars(r), token, stringTenantID)
+	if err != nil {
+		context.LogError(err, microappLog.MessageUnableToFindURLResource)
+		microappWeb.RespondError(w, err)
+		return
+	}
+
+	if err := controller.checkAndInitializeSettingsMetadata(); err != nil {
+		microappWeb.RespondError(w, err)
+		return
+	}
+
+	settingsmetadata := controller.settingsMetadatas
+	if tenantID.String() == "00000000-0000-0000-0000-000000000000" {
+		settingsmetadata = controller.globalsettingsMetadatas
+	}
+
+	tenant, err := controller.getTenant(context, uow, controller.repository, tenantID)
+	if err != nil {
+		microappWeb.RespondError(w, err)
+		return
+	}
+
+	if err = tenant.Update(reqDTO.Settings, settingsmetadata); err != nil {
+		context.LogError(err, microappLog.MessageNewEntityError)
+		microappWeb.RespondError(w, err)
+		return
+	}
+
+	if tenant.Settings != "{}" {
+		queryProcessor := []repository.QueryProcessor{repository.Filter("id = ?", tenantID)}
+		err = controller.repository.Upsert(uow, &tenant, queryProcessor)
+		if err != nil {
+			context.LogError(err, microappLog.MessageUpdateEntityError)
+			microappWeb.RespondError(w, err)
+			return
+		}
+	} else {
+		err = controller.repository.Delete(uow, tenantModel.TenantSettings{}, tenantID)
```

Comment:
This has to be a permanent delete instead of a soft delete

Reply:
changed to DeletePermenant